### PR TITLE
Add org.freeciv.gtk4

### PR DESCRIPTION
**Pull request by freeciv-github**
In a recent poll about freeciv flatpak features,
request for a new flatpak for gtk4-client came out top.

Next new client will be sdl3-client in 2026 release of freeciv-3.2.

### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  Permissions to ~/.freeciv are so that various freeciv modpacks can share some data (e.g. installed modpack) there. The same permission is already given to all existing freeciv flatpaks.
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: org.freeciv.gtk4.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+sdk: org.gnome.Sdk
+command: freeciv-gtk4
+rename-icon: freeciv-client
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=~/.freeciv:create
```
Comment (bbhtt) [inline org.freeciv.gtk4.yml:13]: Why aren't you using `persist=.freeciv` here?

**Comment by bbhtt** [comment]
bot, build org.freeciv.gtk4

File: org.freeciv.gtk4.yml (hunk 1 above)
Comment (freeciv-github) [reply org.freeciv.gtk4.yml:13]: I've understood that 'persist' creates location specific for that flatpak, so wouldn't work for the goal of sharing data between freeciv.org flatpaks.

Comment (bbhtt) [reply org.freeciv.gtk4.yml:13]: Yea sharing configs with other apps won't work.

**Comment by bbhtt** [comment]
bot, build org.freeciv.gtk4

**Pull request closed without merging**